gemrb/GUIScripts/GUIOPT.py

```diff
@@ -58,81 +52,17 @@
 	# just an alias to keep our logic from being plagued by too many GameCheck.IsBG1() checks
 	HelpTextArea2 = HelpTextArea
 
-if GameCheck.IsIWD2():
-	WIDTH = 800
-	HEIGHT = 600
-else:
-	WIDTH = 640
-	HEIGHT = 480
-
-###################################################
-def CloseOptionsWindow ():
-	global GameOptionsWindow, OptionsWindow, PortraitWindow, PauseState
-	global OldPortraitWindow, OldOptionsWindow
-
-	if GameOptionsWindow == None:
-		return
-
-	if GameOptionsWindow:
-		GameOptionsWindow.Unload ()
-	if OptionsWindow:
-		OptionsWindow.Unload ()
-	if PortraitWindow:
-		PortraitWindow.Unload ()
-
-	GameOptionsWindow = None
-	GemRB.SetVar ("OtherWindow", -1)
-	GUICommon.GameWindow.SetVisible(WINDOW_VISIBLE)
-	GemRB.UnhideGUI ()
-	GUICommonWindows.OptionsWindow = OldOptionsWindow
-	OldOptionsWindow = None
-	if not GameCheck.IsBG1():
-		GUICommonWindows.PortraitWindow = OldPortraitWindow
-		OldPortraitWindow = None
-
-	GemRB.GamePause (PauseState, 3)
-	return
-
 ###################################################
-def OpenOptionsWindow ():
+def InitOptionsWindow (Window):
 	"""Open main options window"""
 
-	global GameOptionsWindow, OptionsWindow, PortraitWindow, PauseState
-	global OldPortraitWindow, OldOptionsWindow
-
-	if GUICommon.CloseOtherWindow (OpenOptionsWindow):
-		CloseOptionsWindow()
-		return
-
-	PauseState = GemRB.GamePause (3, 1)
-	GemRB.GamePause (1, 3)
-
 	CommonWindow.CloseContainerWindow ()
-	GemRB.HideGUI ()
-	GUICommon.GameWindow.SetVisible(WINDOW_INVISIBLE)
-
-	GemRB.LoadWindowPack ("GUIOPT", WIDTH, HEIGHT)
-	GameOptionsWindow = Window = GemRB.LoadWindow (2)
-	GemRB.SetVar ("OtherWindow", GameOptionsWindow.ID)
-
-	if OldOptionsWindow == None:
-		# OptionsWindow is the leftmost menu bar window present in most of the games
-		OldOptionsWindow = GUICommonWindows.OptionsWindow
-		OptionsWindow = GemRB.LoadWindow (0)
-		GUICommonWindows.MarkMenuButton (OptionsWindow)
-		GUICommonWindows.SetupMenuWindowControls (OptionsWindow, 0, OpenOptionsWindow)
-		OptionsWindow.SetFrame ()
-		if not GameCheck.IsBG1(): #not in PST either, but it has its own OpenOptionsWindow()
-			OptionsWindow.SetFrame ()
-			#saving the original portrait window
-			OldPortraitWindow = GUICommonWindows.PortraitWindow
-			PortraitWindow = GUICommonWindows.OpenPortraitWindow (0)
 
 	# Return to Game
 	Button = Window.GetControl (11)
 	Button.SetText (10308)
-	Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, OpenOptionsWindow)
-	Button.SetFlags (IE_GUI_BUTTON_CANCEL, OP_OR)
+	Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: Window.Close())
```

Comment:
Codacy found an issue: [Lambda may not be necessary](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)